### PR TITLE
sanity

### PR DESCRIPTION
**Pull request by mjhossen**
## Description

I wrote a python script to find missing files or plotting no of obs or min/max value of a variable, such as absolute dynamic topography. Please check if there is an issue.

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
Thanks @mjhossen . I'll try to look at it later today, but as a start: Remove the templated text of the pr, or fill it in.

**Comment by ShastriPaturi** [comment]
@mjhossen please add reviewers in the top right.

File: mjh_sanity/godas_obssanitycheck.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        help="ioda groups [ObsError, ombg, oman, ObsValue, ...]",
+        type=str, required=True)
+    
+    args = parser.parse_args()
+    provider='jcsda_soca'
+    exp='benchmark_v2'
+    if args.path == None:
+        r2d2_root = '/work/noaa/marine/marineda/r2d2/obs/'
+        print('base r2d2 dir:',r2d2_root)
+    #print(args.file_descriptor)
```
Comment (ShastriPaturi) [inline mjh_sanity/godas_obssanitycheck.py]: Please remove commented lines

**Review by ShastriPaturi — CHANGES_REQUESTED** [review]
Otherwise looks good.
Thanks

File: mjh_sanity/Readme_sc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Use the following api to check missing files or plotting number of obs and min/max value:
```
Comment (guillaumevernieres) [inline mjh_sanity/Readme_sc]: make use of markdown syntax. The convention is to name this file README.md

File: mjh_sanity/godas_obssanitycheck.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        metadata_grp=ncfile.groups['MetaData']
+        var_grp=ncfile.groups[args.group]
+        var_tmp=np.squeeze(var_grp.variables[args.variable][:])
+        minval=min(var_tmp)
+        maxval=max(var_tmp)
+        ncfile.close()
+        return minval, maxval
+        
+if __name__ == '__main__':
+    description = """ Ex: godas_obssanitycheck.py -f adt_c2 -o missing -g ObsValue
```
Comment (guillaumevernieres) [inline mjh_sanity/godas_obssanitycheck.py]: the example doesn't fit the arguments below. -s -e are required for example ...

File: mjh_sanity/godas_obssanitycheck.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        type=str, required=True)
+    parser.add_argument(
+        '-e',
+        '--end',
+        help="end date in the format yearmmdd",
+        type=str, required=True)
+    parser.add_argument(
+        '-g',
+        '--group',
+        help="ioda groups [ObsError, ombg, oman, ObsValue, ...]",
```
Comment (guillaumevernieres) [inline mjh_sanity/godas_obssanitycheck.py]: we're only checking the obs, not the DA. Remove `ombg`, `oma`. But more importantly, why should the user need to know what group to use? The group should not be an argument, remove that option.

File: mjh_sanity/godas_obssanitycheck.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                  """
+    # Command line argument
+    parser = ArgumentParser(
+        description=description,
+        formatter_class=ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-f',
+        '--file_descriptor',
+        type=str, nargs='+',required=True)
+    parser.add_argument(
```
Comment (guillaumevernieres) [inline mjh_sanity/godas_obssanitycheck.py]: might be ok as a first go, but I'd rather not have to provide a variable. You could get that info from the file without asking the user the variable name.

File: mjh_sanity/godas_obssanitycheck.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        '-g',
+        '--group',
+        help="ioda groups [ObsError, ombg, oman, ObsValue, ...]",
+        type=str, required=True)
+    
+    args = parser.parse_args()
+    provider='jcsda_soca'
+    exp='benchmark_v2'
+    if args.path == None:
```
Comment (guillaumevernieres) [inline mjh_sanity/godas_obssanitycheck.py]: just add that as a default value to path L132?

File: mjh_sanity/godas_obssanitycheck.py [hunk 7]
```diff
@@ -0,0 +1,167 @@
+from datetime import datetime
+from netCDF4 import Dataset, num2date, date2num
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+from datetime import datetime, timedelta
+from os.path import exists
+import matplotlib.pyplot as plt
+import numpy as np
+class Sanity:
```
Comment (guillaumevernieres) [inline mjh_sanity/godas_obssanitycheck.py]: You're doing a sanity check, might as well call your class that as well. `SanityCheck` or `ObsSanityCheck` or `ObsCheck` ... Just something more meaningful than `Sanity`.

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
Thanks for doing that work @mjhossen .
1 - find a better name than `mjh_sanity`. I would suggest something like `obsdiag`, `obscheck` for the name of the folder
2 - Do not EVER commit binary files, unless there is a process in place to handle those, which we don't have in this repo. Remove  the png file please.
3 - Tidy your code a bit, remove the debugging comments for example.

I've added a few comments here and there in the code as well.

**Comment by mjhossen** [comment]
I have edited the script accordingly.

**Comment by guillaumevernieres** [comment]
@mjhossen , did you mean to close your PR of was it an unfortunate mouse event?

**Review by guillaumevernieres — APPROVED** [review]
:+1: 
Good enough for now. Thanks @mjhossen

File: obssanitycheck/godas_obssanitycheck.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+            for tx in missing_dates:
+                plt.text(xstep, ystep, '%s'%tx, fontsize=8)
+                ystep=ystep-0.05
+                if ystep < .005: 
+                    xstep=xstep+0.33
+                    ystep=0.95
+        
+        ax.set_xticklabels([])
+        ax.set_yticklabels([])
+        plt.suptitle("File descriptor: %s"%file_descriptor, y=.95);
```
Comment (ShastriPaturi) [inline obssanitycheck/godas_obssanitycheck.py:88]: I would like the "File Descriptor" removed for the title.
For now it is good.

**Review by ShastriPaturi — APPROVED** [review]
Looks good.
Thanks.

File: obssanitycheck/godas_obssanitycheck.py (hunk 8 above)
Comment (mjhossen) [reply obssanitycheck/godas_obssanitycheck.py:88]: Removed.

**Pull request merged**